src/generateSdk/generateSdkApi.ts

```diff
@@ -55,12 +62,30 @@ export async function sdkGeneratorApiHandler(
             );
         }
 
+        if (astGeneratorOutput.program.body !== undefined) {
+            const astClassDefinition = astGeneratorOutput.program.body.find(
+                (node) =>
+                    node.type === AstNodeType.ClassDefinition &&
+                    (node as ClassDefinition).name === classConfiguration.name,
+            ) as ClassDefinition;
+            const currentInvalidHttpMethods = checkInvalidHttpMethods(
+                classConfiguration,
+                astClassDefinition,
+                language,
+            );
+            invalidHttpMethods = invalidHttpMethods.concat(currentInvalidHttpMethods.methods);
+            invalidHttpMethodsString += currentInvalidHttpMethods.methodsString;
+        }
+
```

Comment:
This makes the function a bit hard to read. I propose to refactor the `checkInvalidHttpMethods` function to integrate this logic and check the HTTP method for that class, and also throw a UserError if it finds any problems. I think it Is ok to throw there and show the errors for just one class at a time.
If the function finds no problems, it should just return void and continue execution.